As with SciPy, matplotlib abuses specifications of *build requirements* as a
means to control the environment under which wheels are built for distribution.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -16,7 +16,7 @@
     #      and disabling build isolation.
     #   3. The <2.3 upper bound is for matching the numpy deprecation policy,
     #      it should not be loosened.
-    "numpy>=2.0.0rc1,<2.3",
+    "numpy",
 ]
 
 [tool.setuptools_scm]
